tests/forms_tests/field_tests/test_urlfield.py

```diff
@@ -155,8 +159,41 @@ class URLFieldAssumeSchemeDeprecationTest(FormFieldAssertionsMixin, SimpleTestCa
     def test_urlfield_raises_warning(self):
         msg = (
             "The default scheme will be changed from 'http' to 'https' in Django 6.0. "
-            "Pass the forms.URLField.assume_scheme argument to silence this warning."
+            "Pass the forms.URLField.assume_scheme argument to silence this warning, "
+            "or set the FORMS_URLFIELD_ASSUME_HTTPS transitional setting to True to "
+            "opt into using 'https' as the new default scheme."
         )
         with self.assertWarnsMessage(RemovedInDjango60Warning, msg):
             f = URLField()
             self.assertEqual(f.clean("example.com"), "http://example.com")
+
+    @ignore_warnings(category=RemovedInDjango60Warning)
+    def test_urlfield_forms_urlfield_assume_https(self):
+        with self.settings(FORMS_URLFIELD_ASSUME_HTTPS=True):
+            f = URLField()
+            self.assertEqual(f.clean("example.com"), "https://example.com")
+            f = URLField(assume_scheme="http")
+            self.assertEqual(f.clean("example.com"), "http://example.com")
+
+    def test_override_forms_urlfield_assume_https_setting_warning(self):
+        msg = FORMS_URLFIELD_ASSUME_HTTPS_DEPRECATED_MSG
+        with (
+            self.assertRaisesMessage(RemovedInDjango60Warning, msg),
+            self.settings(FORMS_URLFIELD_ASSUME_HTTPS=True),
+        ):
+            pass
```

Comment:
This line is confusing, could you add a comment explaining why `pass`ing is actually testing something? (I personally don't understand what this is testing :-))

Reply:
> This line is confusing, could you add a comment explaining why `pass`ing is actually testing something? (I personally don't understand what this is testing :-))

It tests that changing a setting via `self.settings()` raises a deprecation warning.

Reply:
Do you think nested context processors are clearer?
```python
with self.assertRaisesMessage(RemovedInDjango60Warning, msg):
    with self.settings(FORMS_URLFIELD_ASSUME_HTTPS=True):
        pass
```

Reply:
Let me check the source code because I was expecting that `self.setting` could be used like a "normal sentence" and not necessarily a context manager:

```python
with self.assertRaisesMessage(RemovedInDjango60Warning, msg):
    self.settings(FORMS_URLFIELD_ASSUME_HTTPS=True)
```

Reply:
My suggestion does not work. Nested withs are a bit clearer, could we add a small comment nest to the `pass` sentence?

```suggestion
            pass  # Changing a setting via self.settings() raises a deprecation warning.
```

Reply:
Sure if you think it's worth doing.